databases/jdp/database.go

```diff
@@ -679,22 +706,80 @@ func (db *Database) Resources(fileIds []string) ([]frictionless.DataResource, er
 		if !found {
 			return nil, &FileIdNotFoundError{fileIds[i]}
 		}
+		id := "JDP:" + md.Id
+		filePath := filepath.Join(strings.TrimPrefix(md.Source.FilePath, filePathPrefix), md.Source.FileName)
+		pi := md.Source.Metadata.Proposal.PI
 		resources[index] = frictionless.DataResource{
-			Id:     "JDP:" + md.Id,
-			Name:   dataResourceName(md.Source.FileName),
-			Path:   filepath.Join(strings.TrimPrefix(md.Source.FilePath, filePathPrefix), md.Source.FileName),
-			Bytes:  md.Source.FileSize,
-			Hash:   md.Source.MD5Sum,
-			Credit: creditFromIdAndMetadata("JDP:"+md.Id, md.Source.Metadata),
+			Id:    id,
```

Comment:
what's the reason for this code that is very similar to the code above? Could you just create a function that extracts the credit from a load of metadata, and then in some future iteration that function can be converted to a call to the credit engine?

Reply:
Great question. Unfortunately, the data structures for the JDP's metadata endpoint and its search endpoint are slightly different, so the function might have quite a lot of parameters in it. But if the JDP team has added a "non-aggregated" version of the search function, I could probably consolidate things. I'll check on this.

Reply:
I ended up consolidating anyway by just creating one datastructure from the other.